### PR TITLE
feat: 🎸 [JIRA:HCPSDKFIORIUIKIT-2880] ActivityItemModel Refactor

### PR DESCRIPTION
**Pull request by hengyi-zhang**

### CODE REVIEW

**Review by dyongxu — CHANGES_REQUESTED** [review]
I saw in the FioriIcons/Resources, some files were updated, (Contents.json).
Usually, we don't update these files under Resource folder, since these files are delivered by Designer team.  
Is there some specific reason why we have to update them here ?

File: Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift [hunk 1]
```diff
@@ -726,3 +728,10 @@ protocol _ProgressIndicatorComponent: _ProgressIndicatorProtocol {}
 /// ```
 // sourcery: CompositeComponent
 protocol _ProcessingIndicatorComponent: _OptionalTitleComponent {}
+
```
Comment (angiexyang) [inline Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift:731]: Could you add some documentation and example code usage?

**Comment by hengyi-zhang** [comment]
> I saw in the FioriIcons/Resources, some files were updated, (Contents.json). Usually, we don't update these files under Resource folder, since these files are delivered by Designer team. Is there some specific reason why we have to update them here ?

fixed!

File: Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift (hunk 1 above)
Comment (hengyi-zhang) [reply Sources/FioriSwiftUICore/_ComponentProtocols/CompositeComponentProtocols.swift:731]: fixed!

**Comment by xiaoqinggrace** [comment]
For the generated code, we do not need to commit if there is no change even if the order of the generated code is different.

File: Sources/FioriSwiftUICore/Models/ModelDefinitions.swift [hunk 2]
```diff
@@ -97,7 +97,11 @@ public protocol ActionModel: ActionComponent {}
 public protocol TextInputModel: TextInputComponent {}
 
 // sourcery: generated_component
-public protocol ActivityItemModel: IconComponent, SubtitleComponent {}
+public protocol _ActivityItemModel: IconComponent, SubtitleComponent {}
+
+/// Deprecated ActivityItemModel
+@available(*, unavailable, renamed: "_ActivityItemModel", message: "Will be removed in the future release. Please create ActivityItemModel with other initializers instead.")
```
Comment (dyongxu) [inline Sources/FioriSwiftUICore/Models/ModelDefinitions.swift:103]:
```suggestion
@available(*, unavailable, renamed: "_ActivityItemModel", message: "Will be removed in the future release. Please create ActivityItem with other initializers instead.")
```

**Review by billzhou0223 — APPROVED** [review]

**Review by KevinZK — APPROVED** [review]

File: Sources/FioriSwiftUICore/Models/ModelDefinitions.swift (hunk 2 above)
Comment (hengyi-zhang) [reply Sources/FioriSwiftUICore/Models/ModelDefinitions.swift]: fixed!

**Review by dyongxu — APPROVED** [review]

**Pull request merged**